### DISCUSSION TITLE
Does the project support a simple compress/decompress interface?

### DISCUSSION THREAD

**Discussion by wqshr12345** [klauspost/compress#879] [Q&A] [answered]
e.g.
Compress(dst, src []byte) ([]byte, error)
Decompress(dst, src []byte) ([]byte, error)
CompressLevel(dst, src []byte, level int) ([]byte, error)
I've checked the interfaces provided by zstd, and I haven't seen any similar interfaces.

**Comment by klauspost** [comment]
https://github.com/klauspost/compress/tree/master/zstd#blocks

https://pkg.go.dev/github.com/klauspost/compress/zstd#Encoder.EncodeAll
https://pkg.go.dev/github.com/klauspost/compress/zstd#Decoder.DecodeAll

**Reply by wqshr12345** [reply]
Does this method support acceleration using multithreading?Thank you.

**Reply by klauspost** [reply, marked as answer]
No. These are strictly single threaded. Use the streaming API for that.
